toolbox/include/base/Parameters.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef WBT_PARAMETERS_H
+#define WBT_PARAMETERS_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace wbt {
+    template <typename T>
+    class Parameter;
+    class ParameterMetadata;
+    class Parameters;
+    const int PARAM_INVALID_INDEX = -1;
+    const std::string PARAM_INVALID_NAME = {};
+} // namespace wbt
+
+/**
+ * @brief Class for storing block's parameters
+ *
+ * This class can contain scalar and vector parameters of the supported types.
+ *
+ * @see wbt::Parameter, wbt::ParameterMetadata, wbt::ParameterType
+ */
+class wbt::Parameters
+{
+public:
+    using ParamIndex = int;
```

Comment:
For indexing purposes, if you need an unsigned quantity I would go for `std::ptrdiff_t`, see http://eigen.tuxfamily.org/index.php?title=3.3#Index_typedef .